x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/dashboards/api/util/installation.ts

```
@@ -0,0 +1,163 @@
+}
+
+export const installTranslated = async ({
+  migrationId,
+  ids,
+  contentManagement,
+  securitySolutionContext,
+  request,
+  context,
+}: InstallTranslatedProps): Promise<number> => {
+  const dashboardMigrationsClient = securitySolutionContext.siemMigrations.getDashboardsClient();
+
+  let installedCount = 0;
+  const installationErrors: Error[] = [];
+
+  // Get installable dashboard migrations
+  const dashboardBatches = dashboardMigrationsClient.data.items.searchBatches(migrationId, {
+    filters: { ids, installable: true },
+  });
+
+  let dashboardsToInstall = await dashboardBatches.next();
+  while (dashboardsToInstall.length) {
+    const { dashboardsToUpdate, errors } = await installDashboards(
+      dashboardsToInstall,
+      contentManagement,
+      request,
+      context
+    );
+    installedCount += dashboardsToUpdate.length;
+    installationErrors.push(...errors);
+    await dashboardMigrationsClient.data.items.update(dashboardsToUpdate);
+    dashboardsToInstall = await dashboardBatches.next();
+  }
+
+  // Throw an error if needed
+  if (installationErrors.length) {
+    throw new Error(installationErrors.map((err) => err.message).join(', '));
+  }
+
+  return installedCount;
+};
+
+const installDashboards = async (
+  dashboardsToInstall: DashboardMigrationDashboard[],
+  contentManagement: ContentManagementServerSetup,
+  request: KibanaRequest,
+  context: RequestHandlerContext
+): Promise<{
+  dashboardsToUpdate: Array<DashboardMigrationDashboard>;
+  errors: Error[];
+}> => {
+  const errors: Error[] = [];
+  const dashboardsToUpdate: Array<DashboardMigrationDashboard> = [];
+
+  const createDashboardsOutcome = await initPromisePool({
+    concurrency: MAX_DASHBOARDS_TO_CREATE_IN_PARALLEL,
+    items: dashboardsToInstall,
+    executor: async (dashboard) => {
+      try {
+        // Parse the dashboard data (assuming it's JSON)
+        const dashboardData = JSON.parse(dashboard.elastic_dashboard?.data || '');
```

This statement will fail if `dashboard.elastic_dashboard?.data` is ''  or empty because of JSON.parse('').

I think we should simple throw with an error `Nothing to install` or `Invalid Dashboard` if `dashboard.elastic_dashboard?.data` is falsy.